cmd/rollouts-controller/main.go

```diff
@@ -132,13 +138,18 @@ func newCommand() *cobra.Command {
 				trafficSplitVersion,
 				nginxIngressClasses,
 				albIngressClasses)
-
 			// notice that there is no need to run Start methods in a separate goroutine. (i.e. go kubeInformerFactory.Start(stopCh)
 			// Start method is non-blocking and runs all registered informers in a dedicated goroutine.
 			kubeInformerFactory.Start(stopCh)
 			argoRolloutsInformerFactory.Start(stopCh)
 			jobInformerFactory.Start(stopCh)
 
+			// Check if Istio installed on cluster before starting dynamicInformerFactory
+			_, err = dynamicClient.Resource(gvk).List(metav1.ListOptions{})
```

Comment:
Since this code is repeated, I think we should have a DoesIstioExist(client) standalone helper utility function that takes a dynamic client.

Also, to limit the number of objects return, we should say `metav1.ListOptions{Limit:1}`. Otherwise we are listing all VirtualServices in the cluster, which could be expensive, simply for the purposes of checking if istio exists.